samples/net/sockets/echo_client/sample.net.sockets.echo_client.nrf_openthread fails to build

`./scripts/sanitycheck -p nrf52840dk_nrf52840 -s samples/net/sockets/echo_client/sample.net.sockets.echo_client.nrf_openthread`

```
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk-0.11.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - not found
-- Looking for stdint.h
-- Looking for stdint.h - not found
-- Looking for stddef.h
-- Looking for stddef.h - not found
-- Check size of unsigned short
-- Check size of unsigned short - failed
-- Check size of unsigned int
-- Check size of unsigned int - failed
-- Check size of unsigned long
-- Check size of unsigned long - failed
CMake Error at /usr/share/cmake/Modules/TestBigEndian.cmake:50 (message):
  no suitable type found
Call Stack (most recent call first):
  /home/galak/git/modules/lib/openthread/CMakeLists.txt:50 (TEST_BIG_ENDIAN)


-- Configuring incomplete, errors occurred!
See also "/home/galak/git/zephyr/sanity-out/nrf52840dk_nrf52840/samples/net/sockets/echo_client/sample.net.sockets.echo_client.nrf_openthread/CMakeFiles/CMakeOutput.log".
See also "/home/galak/git/zephyr/sanity-out/nrf52840dk_nrf52840/samples/net/sockets/echo_client/sample.net.sockets.echo_client.nrf_openthread/CMakeFiles/CMakeError.log".
```

--- a/cmake/extensions.cmake
+++ b/cmake/extensions.cmake
@@ -1746,3 +1746,30 @@ Relative paths are only allowed with `-D${ARGV1}=<path>`")
     set(${ARGV1} ${${ARGV1}} PARENT_SCOPE)
   endif()
 endfunction()
+
+# Usage:
+#   zephyr_get_targets(<directory> <types> <targets>)
+#
+# Get build targets for a given directory and sub-directories.
+#
+# This functions will traverse the build tree, starting from <directory>.
+# It will read the `BUILDSYSTEM_TARGETS` for each directory in the build tree
+# and return the build types matching the <types> list.
+# Example of types: OBJECT_LIBRARY, STATIC_LIBRARY, INTERFACE_LIBRARY, UTILITY.
+#
+# returns a list of targets in <targets> matching the required <types>.
+function(zephyr_get_targets directory types targets)
+    get_property(sub_directories DIRECTORY ${directory} PROPERTY SUBDIRECTORIES)
+    get_property(dir_targets DIRECTORY ${directory} PROPERTY BUILDSYSTEM_TARGETS)
+    foreach(dir_target ${dir_targets})
+      get_property(target_type TARGET ${dir_target} PROPERTY TYPE)
+      if(${target_type} IN_LIST types)
+        list(APPEND ${targets} ${dir_target})
+      endif()
+    endforeach()
+
+    foreach(directory ${sub_directories})
+        zephyr_get_targets(${directory} "${types}" ${targets})
+    endforeach()
+    set(${targets} ${${targets}} PARENT_SCOPE)
+endfunction()

--- a/west.yml
+++ b/west.yml
@@ -111,7 +111,7 @@ manifest:
       revision: 3f545d76a2e6d1db83a470ccdb5bebd1f484e137
       path: modules/lib/loramac-node
     - name: openthread
-      revision: f23165fde96ae8bd710b08c6d77879465ed92cac
+      revision: 7854c687bf23ce509a65d118cbea0c7011a89a1e
       path: modules/lib/openthread
     - name: segger
       revision: 874d9e9696b00c09f9eeefe839028dc25fe44983

